### ISSUE TITLE
Error in bundle function getParentIdsByChild

### ISSUE THREAD

**Issue by Pigau** [magento/magento2#6618]
I notice some mistakes in Magento\Bundle\Model\ResourceModel\Selection:getParentIdsByChild,
this method doesn't return parent bundle ids as expected.
### Preconditions
1. Magento 2.1.0
### Steps to reproduce
1. Use this method on a product which is used as a bundle item
### Expected result
1. Return bundle product id
### Actual result
1. Return nothing

In my project I rewrite this resource model to use a corrected version:

```

    /**
     * Retrieve array of related bundle product ids by selection product id(s)
     *
     * @param int|array $childId
     * @return array
     */
    public function getParentIdsByChild($childId)
    {
        $connection = $this->getConnection();
        $metadata = $this->metadataPool->getMetadata(ProductInterface::class);
        $select = $connection->select()->distinct(
            true
        )->from(
            $this->getMainTable(),
            ['parent_product_id'] // instead of ['e.entity_id as parent_product_id']
        )->join(
            ['e' => $this->metadataPool->getMetadata(ProductInterface::class)->getEntityTable()],
            // = product_id, not parent_product_id
            'e.' . $metadata->getLinkField() . ' = ' .  $this->getMainTable() . '.product_id',
            []
        )->where(
            'e.entity_id IN(?)',
            $childId
        );

        return $connection->fetchCol($select);
     }
```

**Comment by TommyKolkman** [comment]
I have to second this bug: still there in 2.1.1. @Pigau, did you fix it using a plugin or as a preference in you di.xml?

**Comment by Pigau** [comment]
Hi @TommyKolkman , I fix it with a rewrite (a preference), because it use a fix and for me a plugin is for update or add behaviour to a method. But in this case I don't want to wrong version to be use at all. 
Hope you'll find this helpful

**Comment by magento-engcom-team** [comment]
@Pigau, thank you for your report.
The issue is already fixed in develop branch, 2.2.0

**Comment by huykon** [comment]
I tried to use this function but it will get empty array. It's not return the bundle parent id
